mapedit/chunklst.cc

```diff
@@ -566,6 +567,30 @@ void Chunk_chooser::scrolled(
 	chooser->scroll(newindex);
 }
 
+/*
+ *  Handle a draw scroll event.
+ */
+
+void Chunk_chooser::draw_scrolled(  // For scroll events.
+    GtkEventControllerScroll *self, // The scroll event controller,
+    gdouble dx, gdouble dy, // The scroll motion
+    gpointer data           // ->Chunk_chooser.
+) {
+	ignore_unused_variable_warning(self, dx);
+	auto *chooser = static_cast<Chunk_chooser *>(data);
+	GtkAdjustment *adj = gtk_range_get_adjustment(
+	    GTK_RANGE(chooser->vscroll));
+	const gdouble adj_value  = gtk_adjustment_get_value(adj);
+#if defined(MACOSX) && !defined(XWIN)
+	const gdouble new_unit   = 1.0;
+#else
+	const gdouble adj_pgsize = gtk_adjustment_get_page_size(adj);
+	const gdouble new_unit   = pow(adj_pgsize, 2.0/3.0);
+#endif // MACOSX && !XWIN
```

Comment:
Would you mind extracting these lines into a function so that the #ifdef hell can be confined to only one place and call on all the scroll functions? See other notes.

Reply:
I could do much more. Factorizing everything into the parent class `Object_browser` ( `objbrowse.cc` and `objbrowse.h` ). There remains only the activation of the browser scroll from each daughter class ( minus the Palette browser ), since one need he `GtkDrawingArea` and also since it cannot be set in the `Object_browser` constructor, much too early.